great_expectations/data_context/util.py

```diff
@@ -153,9 +156,14 @@ def file_relative_path(dunderfile, relative_path):
     It is meant to be used like the following:
     file_relative_path(__file__, 'path/relative/to/file')
 
+    This has been modified from Dagster's utils:
     H/T https://github.com/dagster-io/dagster/blob/8a250e9619a49e8bff8e9aa7435df89c2d2ea039/python_modules/dagster/dagster/utils/__init__.py#L34
     """
-    return os.path.join(os.path.dirname(dunderfile), relative_path)
+    abs_path = pathlib.Path(source_path).joinpath(relative_path).absolute()
+    str_path = abs_path.as_posix()
+    if strict:
+        assert abs_path.exists(), f"The target relative path {str_path} does not exist!"
```

Comment:
Might want to call `.resolve(strict=strict)` as the first step. Then I think you can skip this logic.
https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve
```python
abs_path = pathlib.Path(source_path).joinpath(relative_path).resolve(strict=strict).absolute()
return abs_path.as_posix()
```

Reply:
BTW I think defaulting to `strict=False` is reasonable (for now), and then you shouldn't have to change it's usage at all.